ndcube/utils/wcs.py

```diff
@@ -119,102 +121,196 @@ def _augment(cls, header, naxis):
-        # this is used to not exceed the range of the item tuple
-        # if the check of the missing_axis which is False if not dead
-        # is a success than the the item of the tuple is added one by
-        # one and if the end of tuple is reached than slice(None, None, None)
-        # is appended.
+        # Create index to track whether the item tuple elements we are dealing with.
         index = 0
-        for _bool in missing_axis:
-            if not _bool:
-                if index is not len(item):
-                    item_checked.append(item[index])
-                    index += 1
-                else:
-                    item_checked.append(slice(None, None, None))
-            else:
+        len_item = len(item)
+        for i, _bool in enumerate(missing_axes_numpy_order):
+            if _bool:
+                # Enter slice(0, 1) for any missing axis.
                 item_checked.append(slice(0, 1))
-        # if all are slice in the item tuple
-        if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
-        # if all are not slices some of them are int then
-        else:
-            # this will make all the item in item_checked as slice.
-            item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
-            for i, it in enumerate(item_checked):
-                if isinstance(it, int):
-                    missing_axis[i] = True
+            else:
+                if index < len_item:
+                    # For non-missing axes with a corresponding sub-item,
+                    # append that item here. If the sub-item is an int,
+                    # convert to appropriate slice for easy identification
+                    # of newly missing/dropped axes later.
+                    if isinstance(item[index], (int, np.int64)):
+                        item_checked.append(slice(item[index], item[index]+1))
+                    elif isinstance(item[index], slice):
+                        item_checked.append(item[index])
+                    else:
+                        raise TypeError("item type at data axis {0} is {1}. ".format(
+                            index, type(item[index]) + "Must be int or slice."))
+                else:
+                    # Subsequent non-missing axes did not have a corresponding slice item
+                    # and are therefore not be sliced.
+                    item_checked.append(slice(None, None))
+                index += 1
+        item_ = tuple(item_checked)
+
+    # Case where item is an an invalid format.
+    else:
+        raise TypeError("item type is {0}.  ".format(type(item)) +
+                        "Must be int, slice, or tuple of ints and/or slices.")
+
+    # Now item_ has an entry for all axes, missing and non-missing,
+    # it can be safely reverse to WCS order
+    # which makes extracting real world coordinates of newly missing/dropped axes easier.
+    item_wcs_order = item[::-1]
```

Comment:
I'm pretty sure this should be ```item_wcs_order = item_[::-1]```.  In fact, I suggest an edit to make things clearer:
```python
item_wcs_order =tuple(item_checked[::-1])
```
and remove the definitions of ```item_``` altogether [here](https://github.com/sunpy/ndcube/pull/151/commits/e2dfd2620933fdf85df7f17ce9a1ddac79f3d3af#diff-b324a6b383c9ce080bf07dcbacda10ddR209) and [here](https://github.com/sunpy/ndcube/pull/151/commits/e2dfd2620933fdf85df7f17ce9a1ddac79f3d3af#diff-b324a6b383c9ce080bf07dcbacda10ddR237)